pkg/api/v2/reporter/reporter.go

```diff
@@ -155,21 +155,28 @@ func (r *reporter) WriteReports(ctx context.Context, resourceErrs ResourceReport
 			merr = multierror.Append(merr, err)
```

Comment:
it would also be prudent to add a comment to this function that mentions that `ResourceReports` will be modified with the results after writing the reports, noting that there may be fewer resources that originally requested if they didn't exist